modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/DatasetFragmentRenderer.java

```diff
@@ -0,0 +1,334 @@
+ */
+@Component(service = FragmentRenderer.class)
+public class DatasetFragmentRenderer implements FragmentRenderer {
+
+	@Override
+	public String getCollectionKey() {
+		return "content-display";
+	}
+
+	@Override
+	public String getConfiguration(
+		FragmentRendererContext fragmentRendererContext) {
+
+		return JSONUtil.put(
+			"fieldSets",
+			JSONUtil.putAll(
+				JSONUtil.put(
+					"fields",
+					JSONUtil.putAll(
+						JSONUtil.put(
+							"label", "dataset-view"
+						).put(
+							"name", "itemSelector"
+						).put(
+							"type", "itemSelector"
+						))))
+		).toString();
+	}
+
+	@Override
+	public String getIcon() {
+		return "table";
+	}
+
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "dataset");
+	}
+
+	@Override
+	public void render(
+			FragmentRendererContext fragmentRendererContext,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		try {
+			PrintWriter printWriter = httpServletResponse.getWriter();
+
+			FragmentEntryLink fragmentEntryLink =
+				fragmentRendererContext.getFragmentEntryLink();
+
+			JSONObject jsonObject =
+				(JSONObject)_fragmentEntryConfigurationParser.getFieldValue(
+					getConfiguration(fragmentRendererContext),
+					fragmentEntryLink.getEditableValues(),
+					fragmentRendererContext.getLocale(), "itemSelector");
+
+			String externalReferenceCode = jsonObject.getString(
+				"externalReferenceCode");
+
+			ObjectEntry fdsViewObjectEntry = null;
+
+			ObjectDefinition fdsViewObjectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					fragmentEntryLink.getCompanyId(), "C_FDSView");
+
+			if (externalReferenceCode != StringPool.BLANK) {
+				fdsViewObjectEntry = _getObjectEntry(
+					fragmentEntryLink.getCompanyId(), externalReferenceCode,
+					fdsViewObjectDefinition);
+			}
+
+			if ((fdsViewObjectEntry == null) &&
+				fragmentRendererContext.isEditMode()) {
+
+				printWriter.write(
+					StringBundler.concat(
+						"<div class=\"portlet-msg-info\">",
+						_language.get(
+							httpServletRequest, "select-a-dataset-view"),
+						"</div>"));
+			}
+
+			if (fdsViewObjectEntry == null) {
+				return;
+			}
+
+			printWriter.write(
+				_buildFragmentHTML(
+					fdsViewObjectEntry, fdsViewObjectDefinition,
+					fragmentRendererContext, httpServletRequest));
+		}
+		catch (Exception exception) {
+			throw new IOException(exception);
+		}
+	}
+
+	private String _buildFragmentHTML(
+			ObjectEntry fdsViewObjectEntry,
+			ObjectDefinition fdsViewObjectDefinition,
+			FragmentRendererContext fragmentRendererContext,
+			HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		StringBundler sb = new StringBundler(5);
+
+		sb.append("<div id=\"");
+		sb.append(fragmentRendererContext.getFragmentElementId());
+		sb.append("\" >");
+
+		ComponentDescriptor componentDescriptor = new ComponentDescriptor(
+			"{FrontendDataSet} from frontend-data-set-web",
```

Comment:
ESM :heart_eyes:

Reply:
yes, and it works like a charm. The only caveat I found here is the position inline, which had to be set to `true` for this to work

Reply:
mmm, do we know why is that? :thinking: